### PR TITLE
feat: firebase sign

### PR DESCRIPTION
**Pull request by Gosh888**

### CODE REVIEW

**Review by edgarmihranyan7 — APPROVED** [review]

File: src/api/auth/auth.service.ts [hunk 1]
```diff
@@ -1,22 +1,34 @@
...
   async singIn(signInDto: SignInDto) {
     const { email, password } = signInDto;
-    return this.clientService.signInAndSendVerification(email, password);
+    return this.firebaseClientService.signInAndSendVerification(
+      email,
+      password,
+    );
+  }
+
+  async refreshToken(refreshToken: string) {
```
Comment (Gosh888) [inline src/api/auth/auth.service.ts:36]: did frontend send refresh token or hi send access token ?

File: src/api/auth/dto/refresh-token.dto.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { IsNotEmpty, IsString } from 'class-validator';
+import { ApiProperty } from '@nestjs/swagger';
+
+export class RefreshTokenDto {
+  @ApiProperty()
+  @IsNotEmpty()
+  @IsString()
+  refreshToken: string;
```
Comment (Gosh888) [inline src/api/auth/dto/refresh-token.dto.ts:8]: smae access token , need to check from you

File: src/api/auth/dto/sign-in.dto.ts [hunk 3]
```diff
@@ -4,10 +4,11 @@ import { ApiProperty } from '@nestjs/swagger';
 export class SignInDto {
   @ApiProperty()
   @IsEmail()
+  @IsString()
   email: string;
 
   @ApiProperty()
-  @IsString()
   @MinLength(6)
+  @IsString()
```
Comment (Gosh888) [inline src/api/auth/dto/sign-in.dto.ts:18]: need strong password validation

File: src/api/auth/dto/sign-up.dto.ts [hunk 4]
```diff
@@ -4,10 +4,11 @@ import { ApiProperty } from '@nestjs/swagger';
 export class SignUpDto {
   @ApiProperty()
   @IsEmail()
+  @IsString()
   email: string;
 
   @ApiProperty()
-  @IsString()
   @MinLength(6)
+  @IsString()
```
Comment (Gosh888) [inline src/api/auth/dto/sign-up.dto.ts:18]: need strong password validation

File: src/firebase/firebase-admin.service.ts [hunk 5]
```diff
@@ -33,4 +36,18 @@ export class FirebaseAdminService implements OnModuleInit {
       throw new InternalServerErrorException(error.message);
     }
   }
+
+  async refreshToken(refreshToken: string) {
```
Comment (Gosh888) [inline src/firebase/firebase-admin.service.ts:40]: is not there way to do it by sdk  ??

File: src/api/auth/dto/auth-user.dto.ts [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import { Expose, Transform } from 'class-transformer';
 import { IsBoolean, IsString } from 'class-validator';
 
-export class UserDto {
```
Comment (Gosh888) [inline src/api/auth/dto/auth-user.dto.ts]: do you use this dto file ??

File: src/firebase/firebase-client.service.ts [hunk 7]
```diff
@@ -78,7 +78,7 @@ export class FirebaseClientService implements OnModuleInit {
   };
 
   mapUserData(userCredential: UserCredential) {
```
Comment (Gosh888) [inline src/firebase/firebase-client.service.ts]: move the map functionality from firebase service to user service and use in auth service

**Comment by edgarmihranyan7** [comment]
http://agent3.ingoarmenia.am:82/Oktell/getClientIncome?ssn=7110910148&from=2023-12-31&to=2024-12-31

File: src/api/auth/auth.service.ts (hunk 1 above)
Comment (edgarmihranyan7) [reply src/api/auth/auth.service.ts:36]: Frontend have to send only refresh token

**Review by edgarmihranyan7 — APPROVED** [review]

**Pull request merged**